lib/inject.js

```diff
@@ -4,8 +4,14 @@
  * @param  {Array}  parent Array of parent frames; or falsey if top level frame
  * @param  {String} script The script to inject
  * @param  {WebDriver} driver The driver to inject into
+ * @param  {Number} [depth] The recursion depth
  */
-function findFramesAndInject(parent, script, driver) {
+function findFramesAndInject(parent, script, driver, depth) {
+	depth = depth || 0;
```

Comment:
`depth` does handle this case, but if I genuinely have 11 `iframes` nested, then the assumption of 10 as top level depth is invalid here.

A better check would be to build a keyValue map of iframe found, either via the relative parent(s) as the as the key, or appending a custom attribute which denotes, that frame has been found/ script injected already.

Reply:
That's a good point, but a key/value map won't work in all cases either:

```js
const iframe = getIframeSomehow()
const parent = iframe.parentNode
const newFrame = iframe.cloneNode()
newFrame.setAttribute('bananas', true)
parent.replaceChild(newFrame, iframe) // Replace the <iframe> with a new one
```

If the user replaces the `<iframe>`, but keeps its `src` or `id` (or whatever attribute(s) we decide to key off of) the same, the map will assume we've already injected `axe-core` into it and will incorrectly "skip" it.